### ISSUE TITLE
Messages in feed gets last character removed if it's a colon

### ISSUE THREAD

**Issue by FuSoftware** [PHPfox-Official/phpfox-v4-issues#2159]
phpFox 4.6.0

On the global feed, messages that end with a colon get the last character (the colon) deleted. Any other character works fine. So far, it seems it's cropped when loading the feed, not due to the autoload.js. For example, querying the /ajax page returns cropped messages with the missing colon.

Message showing the error :
https://puu.sh/z5kGz.png

The edit box for the message, showing the colon should be here
https://puu.sh/z5kGS.png

DevTools view of the cropped message being directly returned from the /ajax page
http://puu.sh/z5kO8.png

**Comment by alexram1313** [comment]
This issue seems problematic for rendering Twemoji Awesome icons on the activity feed when the class is set in the settings. Simply the code minus the closing colon would appear rather than the emoji icon.

**Comment by FuSoftware** [comment]
We experimented to find where the error originates. It seems it happens every time the `shorten` attribute for a text is used in a template. Below, 2 examples for comments and statuses, and how to reproduce.

# Results
## Culprit Function
- Summary
    - `shorten:200:'feed.view_more':true` in both statuses in comments
- Feed (`PF.Base/module/feed/template/default/block/focus.html.php`) => Line 22
    - Problem : `{$aFeed.feed_status|feed_strip|split:55|max_line|shorten:200:'feed.view_more':true}`
    - Fix : `{$aFeed.feed_status|feed_strip|split:55|max_line}`
- Comment (`PF.Base/module/comment/template/default/block/mini.html.php`) => Line 57
    - Problem : `{$aComment.text|feed_strip|shorten:'300':'comment.view_more':true|split:30|max_line}`
    - Fix : `{$aComment.text|feed_strip|split:30|max_line}`
- Screenshot : http://puu.sh/z90pk.png
- Reference in KB
    - https://docs.phpfox.com/display/FOX4MAN/phpFox+4.6.0 (9th New Feature)
## Steps to reproduce
### Feed
- `PF.Base/module/feed/template/default/block/focus.html.php`
```php
{if isset($aFeed.feed_status) && (!empty($aFeed.feed_status) || $aFeed.feed_status == '0')}
<div class="activity_feed_content_status">
	{if strpos($aFeed.feed_status, '<br />') >= 200}
		{$aFeed.feed_status|feed_strip|shorten:200:'feed.view_more':true|split:55|max_line}
	{else}
		{$aFeed.feed_status|feed_strip|split:55|max_line|shorten:200:'feed.view_more':true}
	{/if}

	<br/><b>Raw status</b><br/>
	{$aFeed.feed_status}

	<br/><b>Fixing strip (Original)</b><br/>
	{$aFeed.feed_status|feed_strip|split:55|max_line|shorten:200:'feed.view_more':true}

	<br/><b>Fixing strip (feed_strip)</b><br/>
	{$aFeed.feed_status|split:55|max_line|shorten:200:'feed.view_more':true}

	<br/><b>Fixing strip (split:55)</b><br/>
	{$aFeed.feed_status|feed_strip|max_line|shorten:200:'feed.view_more':true}

	<br/><b>Fixing strip (max_line)</b><br/>
	{$aFeed.feed_status|feed_strip|split:55|shorten:200:'feed.view_more':true}

	<br/><b>Fixing strip (shorten:200:'feed.view_more':true)</b><br/>
	{$aFeed.feed_status|feed_strip|split:55|max_line}

</div>
{/if}
```
# Feed Status
## Service Feed
- Lookup in the feed class at `PF.Base/module/feed/include/service/feed.class.php` (*Line 1517*)
    - Line 1537 : `Phpfox_Template::instance()->assign(['aFeed' => $aFeeds[0]])->getTemplate('feed.block.entry');`
        - Use of the template `feed.block.entry`
        - Feed retreived from `$aFeeds = Phpfox::getService('feed')->get(null, $iId);`
    - Line 239 to 1013 `public function get($iUserId = null, $iFeedId = null, $iPage = 0, $bForceReturn = false, $bLimit = true, $iLastFeedId = null)`
    - `var_dump($aFeeds)` showed that the statuses are properly stored
## Focus.html.php 
- Lookup in the focus template at `PF.Base/module/feed/template/default/block/focus.html.php`
    - Line 22 : Maybe something in the {if} creates a problem.
        - Problem : `{$aFeed.feed_status|feed_strip|split:55|max_line|shorten:200:'feed.view_more':true}`
        - Fix : `{$aFeed.feed_status|feed_strip|split:55|max_line}`
# Comment
- Lookup in the comment template at `PF.Base/module/comment/template/default/block/mini.html.php`
    - Line 57 : `{$aComment.text|feed_strip|shorten:'300':'comment.view_more':true|split:30|max_line}`

**Comment by neil96** [comment]
Thanks very much @FuSoftware. We confirmed this is a bug and we fixed in in phpFox 4.6.1.

Regards.
